src/main/java/com/hazelcast/web/WebFilterConfig.java

```diff
@@ -0,0 +1,293 @@
+    private String cookieName;
+    private String cookieDomain;
+    private boolean cookieSecure;
+    private boolean cookieHttpOnly;
+    private String cookiePath;
+
+    private WebFilterConfig() {
+    }
+
+    public static WebFilterConfig create(FilterConfig filterConfig, Properties properties) {
+        boolean useClient = getBoolean(filterConfig, properties, USE_CLIENT, false);
+
+        if (useClient) {
+            if (paramExists(filterConfig, properties, SESSION_TTL_CONFIG)) {
+                throw new InvalidConfigurationException(SESSION_TTL_CONFIG + " cannot be used with client mode.");
+            }
+            if (paramExists(filterConfig, properties, CONFIG_LOCATION)) {
+                throw new InvalidConfigurationException(CONFIG_LOCATION + " cannot be used with client mode.");
+            }
+        } else {
+            if (paramExists(filterConfig, properties, CLIENT_CONFIG_LOCATION)) {
+                throw new InvalidConfigurationException(CLIENT_CONFIG_LOCATION + " cannot be used with P2P mode.");
+            }
+        }
+
+        // Client mode parameters
+        String clientConfigLocation = getString(filterConfig, properties, CLIENT_CONFIG_LOCATION, null);
+
+        // P2P mode parameters
+        int sessionTtlSeconds = getInt(filterConfig, properties, SESSION_TTL_CONFIG, SESSION_TTL_DEFAULT_SECONDS);
+        String configLocation = getString(filterConfig, properties, CONFIG_LOCATION, null);
+
+        URL configUrl = validateAndGetConfigUrl(filterConfig.getServletContext(), useClient, configLocation,
+                clientConfigLocation);
+
+        // common parameters
+        String instanceName = getString(filterConfig, properties, INSTANCE_NAME, null);
+        String mapName = getString(filterConfig, properties, MAP_NAME,
+                "_web_" + filterConfig.getServletContext().getServletContextName());
+        boolean stickySession = getBoolean(filterConfig, properties, STICKY_SESSION_CONFIG, true);
+        boolean shutdownOnDestroy = getBoolean(filterConfig, properties, SHUTDOWN_ON_DESTROY, true);
+        boolean deferredWrite = getBoolean(filterConfig, properties, DEFERRED_WRITE, false);
+        boolean useRequestParameter = getBoolean(filterConfig, properties, USE_REQUEST_PARAMETER, false);
+        boolean debug = getBoolean(filterConfig, properties, DEBUG, false);
+        Set<String> transientAttributes = getStringSet(filterConfig, properties, TRANSIENT_ATTRIBUTES);
+        String cookieName = getString(filterConfig, properties, COOKIE_NAME, "hazelcast.sessionId");
+        String cookieDomain = getString(filterConfig, properties, COOKIE_DOMAIN, null);
+        boolean cookieSecure = getBoolean(filterConfig, properties, COOKIE_SECURE, false);
+        boolean cookieHttpOnly = getBoolean(filterConfig, properties, COOKIE_HTTP_ONLY, false);
+        String cookiePath = getString(filterConfig, properties, COOKIE_PATH, null);
+
+        WebFilterConfig wfc = new WebFilterConfig();
```

Comment:
There's a single place where we create `WebFilterConfig` on `WebFilter#init` method. As we advise users to define only a single instance of `WebFilter` in their `web.xml` file, there will effectively be a single `WebFilterConfig` instance and it will be initialised only once. 

Also, we let users configure `WebFilter` via Spring Framework and it wouldn't place nicely with `static` either. 

Another reason is it would make unit testing harder for classes that are using `WebFilterConfig`.